### PR TITLE
feat(keyless-backup): add sign in with email screen

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

[Figma](https://www.figma.com/file/gGp6SVFWMbnSltkGzNzrpC/Wallet-Recovery?type=design&node-id=2698-22743&t=3w7xdC81dvYiLeye-0), depends on #3732

Just the UI, button click will be in a follow up

### Test plan

Manual

### Related issues

- Part of ACT-743

### Backwards compatibility

Yes

### CODE REVIEW

File: src/keylessBackup/SignInWithEmail.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+SignInWithEmail.navigationOptions = () => ({
+  ...emptyHeader,
+  headerLeft: () => (
+    <TopBarIconButton
+      style={styles.cancelButton}
+      icon={<Times height={16} />}
+      onPress={() => {
+        navigateHome({ params: { initialScreen: Screens.Settings } })
```
Comment (jophish) [inline src/keylessBackup/SignInWithEmail.tsx:50]: Do we want an analytics event for this cancel press?

File: src/keylessBackup/SignInWithEmail.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import Times from 'src/icons/Times'
+import { emptyHeader } from 'src/navigator/Headers'
+import { navigateHome } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import fontStyles from 'src/styles/fonts'
+
+function onPressGoogle() {
+  ValoraAnalytics.track(KeylessBackupEvents.sign_in_with_google)
+  // TODO: start google sign-in
```
Comment (jophish) [inline src/keylessBackup/SignInWithEmail.tsx:17]: This will be included in the same ticket, ACT-743, right?

**Review by jophish — COMMENTED** [review]
Looks good to me, just one question about a potential analytics event.

File: src/keylessBackup/SignInWithEmal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { fireEvent, render } from '@testing-library/react-native'
```
Comment (anthonydaoud) [inline src/keylessBackup/SignInWithEmal.test.tsx:1]: There's a typo in the name of this file

File: src/keylessBackup/SignInWithEmail.tsx (hunk 2 above)
Comment (satish-ravi) [reply src/keylessBackup/SignInWithEmail.tsx:17]: yes, will be in a follow up PR

**Review by anthonydaoud — APPROVED** [review]
Looks good!

**Pull request merged**